modules/nw-ptp-configuring-linuxptp-services-as-boundary-clock.adoc

```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// * networking/multiple_networks/configuring-ptp.adoc
+
+[id="configuring-linuxptp-services-as-boundary-clock_{context}"]
+= Configuring `linuxptp` services as Boundary Clock
+
+The PTP Operator adds the `PtpConfig.ptp.openshift.io` custom resource definition (CRD) to {product-title}.
+You can configure the `linuxptp` services (`ptp4l`, `phc2sys`) by creating a `PtpConfig` custom resource (CR) object.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* Install the PTP Operator.
+
+.Procedure
+
+. Create the following `PtpConfig` CR, and then save the YAML in the `boundary-clock-ptp-config.yaml` file.
++
+[source,yaml]
+----
+apiVersion: ptp.openshift.io/v1
+kind: PtpConfig
+metadata:
+  name: boundary-clock-ptp-config <1>
+  namespace: openshift-ptp
+spec:
+  profile: <2>
+  - name: "profile1" <3>
+    interface: "" <4>
+    ptp4lOpts: "-s -2" <5>
+    ptp4lConf: | <6>
+      [ens1f0] <7>
+      masterOnly              0
+
+      [ens1f3] <8> 
+      masterOnly              1
+
+      [global]
+      #
+      # Default Data Set
+      #
+      twoStepFlag 1
+      slaveOnly 0
+      priority1 128
+      priority2 128
+      domainNumber 24
+      #utc_offset 37
+      clockClass 248
+      clockAccuracy 0xFE
+      offsetScaledLogVariance 0xFFFF
+      free_running 0
+      freq_est_interval 1
+      dscp_event 0
+      dscp_general 0
+      dataset_comparison ieee1588
+      G.8275.defaultDS.localPriority 128
+      ...
```

Comment:
IF we include the elipsis here, is this still valid YAML that can be applied to the cluster? I think we should make sure that the YAML can be applied.

Reply:
@kononovn Could you please take a look at this question when you review?

Reply:
yes "..." is not correct value. I have this `ptp4lConf` example:
```
#
# Default Data Set
#
twoStepFlag                       1
#slaveOnly                        1
priority1                         128
priority2                         128
domainNumber                      24
#utc_offset                       37
clockClass                        248
clockAccuracy                     0xFE
offsetScaledLogVariance         0xFFFF
free_running                      0
freq_est_interval               1
dscp_event                        0
dscp_general                      0
dataset_comparison              G.8275.x
G.8275.defaultDS.localPriority  128
#
# Port Data Set
#
logAnnounceInterval          -3
logSyncInterval                -4
logMinDelayReqInterval       -4
logMinPdelayReqInterval      -4
announceReceiptTimeout       3
syncReceiptTimeout           0
delayAsymmetry                 0
fault_reset_interval         4
neighborPropDelayThresh      20000000
masterOnly                     0
G.8275.portDS.localPriority  128
#
# Run time options
#
assume_two_step              0
logging_level                6
path_trace_enabled         0
follow_up_info               0
hybrid_e2e                   0
inhibit_multicast_service  0
net_sync_monitor           0
tc_spanning_tree           0
tx_timestamp_timeout       10
#was 1 (default !)
unicast_listen          0
unicast_master_table  0
unicast_req_duration  3600
use_syslog              1
verbose                   0
summary_interval      -4
kernel_leap             1
check_fup_sync          0
#
# Servo Options
#
pi_proportional_const     0.0
pi_integral_const         0.0
pi_proportional_scale     0.0
pi_proportional_exponent  -0.3
pi_proportional_norm_max  0.7
pi_integral_scale         0.0
pi_integral_exponent      0.4
pi_integral_norm_max      0.3
step_threshold              0
first_step_threshold      0.00002
max_frequency               900000000
clock_servo                 pi
sanity_freq_limit         200000000
ntpshm_segment              0
#
# Transport options
#
transportSpecific  0x0
ptp_dst_mac          01:1B:19:00:00:00
p2p_dst_mac          01:80:C2:00:00:0E
udp_ttl                1
udp6_scope           0x0E
uds_address          /var/run/ptp4l
#
# Default interface options
#
clock_type             OC
network_transport    UDPv4
delay_mechanism        E2E
time_stamping          hardware
tsproc_mode            filter
delay_filter           moving_median
delay_filter_length  10
egressLatency          0
ingressLatency         0
boundary_clock_jbod  1
#
# Clock description
#
productDescription    ;;
revisionData            ;;
manufacturerIdentity  00:00:00
userDescription         ;
timeSource              0xA0 

```